modules/nf-core/deseq2/differential/templates/deseq_de.r

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/env Rscript
+
+################################################
+################################################
+## Functions                                  ##
+################################################
+################################################
+
+#' Parse out options from a string without recourse to optparse
+#'
+#' @param x Long-form argument list like --opt1 val1 --opt2 val2
+#'
+#' @return named list of options and values similar to optparse
+
+parse_args <- function(x){
+    args_list <- unlist(strsplit(x, ' ?--')[[1]])[-1]
+    args_vals <- unlist(lapply(args_list, function(y) strsplit(y, ' +')))
+
+    as.list(structure(args_vals[c(FALSE, TRUE)], names = args_vals[c(TRUE, FALSE)]))
+}
+
+################################################
+################################################
+## PARSE PARAMETERS FROM NEXTFLOW             ##
+################################################
+################################################
+
+# I've defined these in a single array like this so that we could go back to an
+# optparse-driven method in future with module bin/ directories, rather than
+# the template
+
+# Set defaults and classes
+
+opt <- list(
+    count_file = '$counts',
+    sample_file = '$samplesheet',
+    contrast_variable = '$contrast_meta.variable',
+    reference_level = '$contrast_meta.reference',
+    treatment_level = '$contrast_meta.target',
+    blocking_variables = '$contrast_meta.blocking',
+    gene_id_col = "gene_id",
+    sample_id_col = "experiment_accession",
+    test = "Wald",
+    fit_type = "parametric",
+    sf_type = 'ratio',
+    min_replicates_for_replace = 7,
+    use_t = FALSE,
+    lfc_threshold = 0,
+    alt_hypothesis = 'greaterAbs',
+    independent_filtering = 'TRUE',
+    p_adjust_method = 'BH',
+    alpha = 0.1,
+    minmu = 0.5,
+    write_normalised = TRUE,
+    write_variance_stabilised = TRUE,
+    vs_method = 'vst',
+    random_seed = 0,
+    round_results = FALSE
+)
+opt_types <- lapply(opt, class)
+
+# Apply parameter overrides
+
+args_opt <- parse_args('$task.ext.args')
+for ( ao in names(args_opt)){
+    if (! ao %in% names(opt)){
+        stop(paste("Invalid option:", ao))
+    }else{
+        opt[[ao]] <- as(args_opt[[ao]], opt_types[[ao]])
+    }
+}
+
+# Check file inputs are valid
+
+for (file_input in c('count_file', 'sample_file')){
+    if (is.null(opt[[file_input]])) {
+        stop(paste("Please provide", file_input), call. = FALSE)
+    }
+
+    if (! file.exists(opt[[file_input]])){
+        stop(paste0('Value of ', file_input, ': ', opt[[file_input]], ' is not a valid file'))
+    }
+}
+
+if (opt\$random_seed > 0){
```

Comment:
Does the set.seed actually make a difference? I didn't think DESeq used any randomness. Is this just a hangover from the precision issues?

Reply:
You're right, and it didn't help, I'll strip it out.